plugins/autoscroll.koplugin/main.lua

```diff
@@ -0,0 +1,161 @@
+local Device = require("device")
+local Event = require("ui/event")
+local PluginShare = require("pluginshare")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local AutoScroll = WidgetContainer:new{
+    name = 'autoscroll',
+    is_doc_only = true,
+    autoscroll_sec = G_reader_settings:readSetting("autoscroll_timeout_seconds") or 0,
+    autoscroll_distance = G_reader_settings:readSetting("autoscroll_distance") or 1,
+    enabled = G_reader_settings:isTrue("autoscroll_enabled"),
+    settings_id = 0,
+    last_action_sec = os.time(),
+}
+
+function AutoScroll:_enabled()
+    return self.enabled and self.autoscroll_sec > 0
+end
+
+function AutoScroll:_schedule(settings_id)
+    if not self:_enabled() then
+        logger.dbg("AutoScroll:_schedule is disabled")
+        return
+    end
+    if self.settings_id ~= settings_id then
+        logger.dbg("AutoScroll:_schedule registered settings_id ",
+                   settings_id,
+                   " does not equal to current one ",
+                   self.settings_id)
+        return
+    end
+
+    local delay = self.last_action_sec + self.autoscroll_sec - os.time()
+
+    if delay <= 0 then
+        logger.dbg("AutoScroll: go to next page")
+        self.ui:handleEvent(Event:new("GotoViewRel", self.autoscroll_distance))
+        logger.dbg("AutoScroll: schedule at ", os.time() + self.autoscroll_sec)
+        UIManager:scheduleIn(self.autoscroll_sec, function() self:_schedule(settings_id) end)
+    else
+        logger.dbg("AutoScroll: schedule at ", os.time() + delay)
+        UIManager:scheduleIn(delay, function() self:_schedule(settings_id) end)
+    end
+end
+
+function AutoScroll:_deprecateLastTask()
+    PluginShare.pause_auto_suspend = false
+    self.settings_id = self.settings_id + 1
+    logger.dbg("AutoScroll: deprecateLastTask ", self.settings_id)
+end
+
+function AutoScroll:_start()
+    if self:_enabled() then
+        logger.dbg("AutoScroll: start at ", os.time())
+        PluginShare.pause_auto_suspend = true
+        self.last_action_sec = os.time()
+        self:_schedule(self.settings_id)
+
+        local InfoMessage = require("ui/widget/infomessage")
+        UIManager:show(InfoMessage:new{
+            text = T(_("Autoscroll is now active and will automatically turn the page every %1 seconds."), self.autoscroll_sec),
```

Comment:
"turn the page" - but we now can turn some fraction of the page :)
So, just "scroll", or `if distance == 1 then "turn the page" else T("scroll %1%% of the page", distance)` ?